core/src/main/java/org/mule/config/pool/DefaultThreadPoolFactory.java

```diff
@@ -81,8 +84,9 @@ private void configureThreadPoolExecutor(String name, ThreadingProfile tp, Threa
     @Override
     public ScheduledThreadPoolExecutor createScheduledPool(String name, ThreadingProfile tp)
     {
-        ScheduledThreadPoolExecutor pool = internalCreateScheduledPool(tp);
-        configureThreadPoolExecutor(name, tp, pool);
+        ThreadingProfile threadingProfileCopy = tp instanceof DirectThreadingProfile ? tp : new ImmutableThreadingProfile(tp);
```

Comment:
Why can't you create an ImmutableThreadingProfile from a DirectThreadingProfile in order to avoid all these instanceof checks?